client/src/includes/contentMetrics.ts

```diff
@@ -41,9 +41,50 @@ export const getReadingTime = (lang: string, wordCount: number): number => {
   return readingTime;
 };
 
+export const getReadabilityScore = (lang: string, text: string): string => {
+  const segmenter = new Intl.Segmenter(lang, { granularity: 'sentence' });
+  const sentences: Intl.SegmentData[] = Array.from(segmenter.segment(text));
+  const sentenceCount = sentences.length;
+
+  const wordSegmenter = new Intl.Segmenter(lang, { granularity: 'word' });
+  const words: Intl.SegmentData[] = Array.from(wordSegmenter.segment(text));
+  const wordCount = words.reduce(
+    (count, segment) => (segment.isWordLike ? count + 1 : count),
+    0,
+  );
+  // Count long words (6+ characters)
+  const longWordCount = words.reduce((count, segment) => {
+    if (segment.isWordLike && segment.segment.length > 6) {
+      return count + 1;
+    }
+    return count;
+  }, 0);
+
+  // If no sentences found, return "Poor"
+  if (sentenceCount === 0 || wordCount === 0) {
+    return gettext('Good');
+  }
+
+  // LIX formula: (A / B) + (C * 100 / A)
+  // A = number of words, B = number of sentences, C = number of long words
+  const avgWordsPerSentence = wordCount / sentenceCount;
+  const percentLongWords = (longWordCount / wordCount) * 100;
+  const lixScore = avgWordsPerSentence + percentLongWords;
+
+  // Convert numerical score to translated label
+  if (lixScore < 30) {
+    return gettext('Good');
+  } else if (lixScore < 40) {
+    return gettext('Fair');
+  }
+
+  return gettext('Poor');
```

Comment:
I think it would be useful to split this into two  separate functions: one for getting the LIX score, and one for determining the actual label we use for display. It would make it easier to support different score ranges for different languages, and people interested in the value can listen to the `w-preview:content` event and get it from `event.detail.metrics`.

Reply:
refactored!